src/main/java/eu/bbmri_eric/negotiator/negotiation/NegotiationController.java

```diff
@@ -162,17 +164,28 @@ public EntityModel<NegotiationDTO> retrieve(@Valid @PathVariable String id) {
    *
    * @param id of the negotiation
    * @param event from NegotiationEvents
+   * @param negotiationUpdateLifecycleDTO an optional body with details about the event
    * @return NegotiationDTO with updated state if valid
    */
-  @PutMapping("/negotiations/{id}/lifecycle/{event}")
+  @PutMapping(
+      value = "/negotiations/{id}/lifecycle/{event}",
+      consumes = MediaType.APPLICATION_JSON_VALUE)
   public ResponseEntity<?> sendEvent(
-      @Valid @PathVariable String id, @Valid @PathVariable("event") NegotiationEvent event) {
+      @Valid @PathVariable String id,
+      @Valid @PathVariable("event") NegotiationEvent event,
+      @RequestBody(required = false) @Nullable
+          NegotiationUpdateLifecycleDTO negotiationUpdateLifecycleDTO) {
     if (!AuthenticatedUserContext.isCurrentlyAuthenticatedUserAdmin()
         && !isCreator(negotiationService.findById(id, false))) {
       throw new ResponseStatusException(HttpStatus.FORBIDDEN);
     }
     // Process the request
-    negotiationLifecycleService.sendEvent(id, event);
+    String message = null;
+    if (negotiationUpdateLifecycleDTO != null
```

Comment:
Objects.isNull()

Reply:
The documentation says:
This method exists to be used as a [Predicate](https://docs.oracle.com/javase/8/docs/api/java/util/function/Predicate.html), filter(Objects::isNull)